src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java

```diff
@@ -509,41 +503,46 @@ public static boolean goodProbability(final double result) {
     //
 
     public static double logFactorial(final int x) {
-        if (x >= LogFactorialCache.size() || x < 0)
-            return Gamma.logGamma(x + 1);
-        else
-            return LogFactorialCache.get(x);
+        return logFactorialCache.get(x);
     }
 
     /**
-     * Wrapper class so that the logFactorial array is only calculated if it's used
+     * The size of the precomputed cache of logs.
+     * The caches are immutable after creation and so it's no big deal that they are static.
      */
-    private static class LogFactorialCache {
+    private static final int PRECOMPUTED_LOGS = 10_000;
```

Comment:
I'm a bit worried that we might be losing the benefit of the caching.  Do we know that most of the expense is in the 0-10,000 range?  Would we do better by just baking a log table into the jar?